pkg/proxy/creator/creator.go

```diff
@@ -0,0 +1,238 @@
+}
+
+// CreateProxy will create a new instance of proxy
+func CreateProxy(args ArgsProxy) (proxy2.ProxyHandler, error) {
+	proxyInstance := &proxy{
+		closableComponents: data.NewClosableComponentsHandler(),
+	}
+
+	statusMetricsProvider := metrics.NewStatusMetrics()
+
+	nodesProviderFactory, err := observer.NewNodesProviderFactory(*args.Config, "")
+	if err != nil {
+		return nil, err
+	}
+	observersProvider, err := nodesProviderFactory.CreateObservers()
+	if err != nil {
+		return nil, err
+	}
+	fullHistoryNodesProvider, err := nodesProviderFactory.CreateFullHistoryNodes()
+	if err != nil {
+		if !errors.Is(err, observer.ErrEmptyObserversList) {
+			return nil, err
+		}
+	}
+
+	pubKeyConverter := args.NodeHandler.GetCoreComponents().AddressPubKeyConverter()
+	shardCoord := args.NodeHandler.GetShardCoordinator()
+	bp, err := processProxy.NewBaseProcessor(
+		args.Config.GeneralSettings.RequestTimeoutSec,
+		shardCoord,
+		observersProvider,
+		fullHistoryNodesProvider,
+		pubKeyConverter,
+	)
+	if err != nil {
+		return nil, err
+	}
+	bp.StartNodesSyncStateChecks()
+
+	connector := database.NewDisabledElasticSearchConnector()
+	accntProc, err := processProxy.NewAccountProcessor(bp, pubKeyConverter, connector)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO faucet processor is disabled
+	faucetValue := big.NewInt(0)
+	faucetProc, err := processFactory.CreateFaucetProcessor(bp, shardCoord, faucetValue, pubKeyConverter, "")
+	if err != nil {
+		return nil, err
+	}
+
+	txProc, err := processFactory.CreateTransactionProcessor(
+		bp,
+		pubKeyConverter,
+		args.NodeHandler.GetCoreComponents().Hasher(),
+		args.NodeHandler.GetCoreComponents().InternalMarshalizer(),
+		args.Config.GeneralSettings.AllowEntireTxPoolFetch,
+	)
+	if err != nil {
+		return nil, err
+	}
+	scQueryProc, err := processProxy.NewSCQueryProcessor(bp, pubKeyConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	htbCacher := cache.NewHeartbeatMemoryCacher()
+	cacheValidity := time.Duration(args.Config.GeneralSettings.HeartbeatCacheValidityDurationSec) * time.Second
+
+	nodeGroupProc, err := processProxy.NewNodeGroupProcessor(bp, htbCacher, cacheValidity)
+	if err != nil {
+		return nil, err
+	}
+
+	valStatsCacher := cache.NewValidatorsStatsMemoryCacher()
+	cacheValidity = time.Duration(args.Config.GeneralSettings.ValStatsCacheValidityDurationSec) * time.Second
+
+	valStatsProc, err := processProxy.NewValidatorStatisticsProcessor(bp, valStatsCacher, cacheValidity)
+	if err != nil {
+		return nil, err
+	}
+
+	economicMetricsCacher := cache.NewGenericApiResponseMemoryCacher()
+	cacheValidity = time.Duration(args.Config.GeneralSettings.EconomicsMetricsCacheValidityDurationSec) * time.Second
+
+	nodeStatusProc, err := processProxy.NewNodeStatusProcessor(bp, economicMetricsCacher, cacheValidity)
+	if err != nil {
+		return nil, err
+	}
+
+	proxyInstance.closableComponents.Add(nodeGroupProc, valStatsProc, nodeStatusProc, bp)
+
+	//nodeGroupProc.StartCacheUpdate()
```

Comment:
uncomment this?

Reply:
We don't need heartbeats here, so I will remove this commented line.